editor/modes/visual-editor/block.js

```diff
@@ -251,6 +298,18 @@ class VisualEditorBlock extends Component {
 	onKeyDown( event ) {
 		const { keyCode, target } = event;
 
+		this.handleToolbarTabCycle( event );
+		this.handleToolbarArrowCycle( event );
+
+		if ( keyCode === F10 && ! ( event.altKey || event.ctrlKey || event.shiftKey || event.metaKey ) ) {
+			event.preventDefault();
+			event.stopPropagation();
+			if ( this.props.isTyping ) {
+				this.props.onStopTyping();
+			}
+			this.props.onFocus( this.props.uid, { toolbar: true } );
```

Comment:
Curious: Do we need toolbar focus to be a concern of the Redux focus state? I don't feel particularly strongly either way, but it seems like we could achieve the same effect with the component's local state. Unless we're ever moving focus from one block to another and assigning `{ toolbar: true }` in the process (would this be handled well if it were to occur?)